### PR TITLE
update to 0.9.16

### PR DESCRIPTION
**Pull request by stanly-johnson**
Requires https://github.com/open-web3-stack/open-runtime-module-library/issues/686

### CODE REVIEW

**Comment by olanod** [comment]
`0.9.16` is just released isn't it?

File: node/src/command.rs [hunk 1]
```diff
@@ -30,7 +33,7 @@ fn load_spec(id: &str) -> std::result::Result<Box<dyn sc_service::ChainSpec>, St
 
 impl SubstrateCli for Cli {
 	fn impl_name() -> String {
-		"Virto Parachain Collator".into()
+		"Parachain Collator Template".into()
```
Comment (olanod) [inline node/src/command.rs:33]:
```suggestion
		"Virto Parachain Collator".into()
```

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -598,7 +601,10 @@ impl orml_unknown_tokens::Config for Runtime {
 pub struct VirtoDisputeResolver;
 impl virto_payment::DisputeResolver<AccountId> for VirtoDisputeResolver {
 	fn get_origin() -> AccountId {
-		Sudo::key()
+		match Sudo::key() {
```
Comment (olanod) [inline runtime/src/lib.rs:604]: Is this returning an `Option` a new thing? I think it's valid to `expect` that the sudo key is set

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -610,7 +616,11 @@ impl virto_payment::FeeHandler<Asset, Balance, AccountId, BlockNumber> for Virto
 		_remark: &virto_payment::PaymentDetail<Asset, Balance, AccountId, BlockNumber>,
 	) -> (AccountId, Percent) {
 		const VIRTO_MARKETPLACE_FEE_PERCENT: Percent = Percent::from_percent(0);
-		(Sudo::key(), VIRTO_MARKETPLACE_FEE_PERCENT)
+		let fee_receiver = match Sudo::key() {
```
Comment (olanod) [inline runtime/src/lib.rs:619]: ☝🏽

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -53,7 +53,7 @@ where
 /// Generate the session keys from individual elements.
 ///
 /// The input must be a tuple of individual keys (a single arg for now since we have just one key).
-pub fn virto_session_keys(keys: AuraId) -> virto_runtime::SessionKeys {
+pub fn template_session_keys(keys: AuraId) -> virto_runtime::SessionKeys {
```
Comment (olanod) [inline node/src/chain_spec.rs:56]:
```suggestion
pub fn virto_session_keys(keys: AuraId) -> virto_runtime::SessionKeys {
```

File: node/src/chain_spec.rs [hunk 5]
```diff
@@ -159,7 +160,9 @@ pub fn local_testnet_config() -> ChainSpec {
 		// Telemetry
 		None,
 		// Protocol ID
-		Some("virto-local"),
+		Some("template-local"),
```
Comment (olanod) [inline node/src/chain_spec.rs:163]:
```suggestion
		Some("virto-local"),
```

File: node/src/chain_spec.rs [hunk 6]
```diff
@@ -195,9 +198,9 @@ fn testnet_genesis(
 				.into_iter()
 				.map(|(acc, aura)| {
 					(
-						acc.clone(),              // account id
-						acc,                      // validator id
-						virto_session_keys(aura), // session keys
+						acc.clone(),                 // account id
+						acc,                         // validator id
+						template_session_keys(aura), // session keys
```
Comment (olanod) [inline node/src/chain_spec.rs:203]:
```suggestion
						virto_session_keys(aura), // session keys
```

**Review by olanod — COMMENTED** [review]
Thanks for this, just need to check the mentions of "template"

File: runtime/src/lib.rs (hunk 2 above)
Comment (stanly-johnson) [reply runtime/src/lib.rs]: Yes the sudo key being optional is part of the update of 0.9.16. Changed to expect a sudo key always

**Review by olanod — APPROVED** [review]

**Review by olanod — APPROVED** [review]

**Pull request merged**